invenio_rdm_records/services/secret_links/service.py

```diff
@@ -0,0 +1,298 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2021 CERN.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""RDM Secret Link Service."""
+
+from datetime import datetime
+
+import arrow
+from flask_babelex import lazy_gettext as _
+from invenio_db import db
+from invenio_drafts_resources.services.records import RecordService
+from invenio_records_resources.services.records.schema import \
+    ServiceSchemaWrapper
+from marshmallow.exceptions import ValidationError
+
+
+class SecretLinkService(RecordService):
```

Comment:
inheritance/design: `SecretLinkService` inherits from `RecordsService` which after a few parental relations end up in [here](https://github.com/inveniosoftware/invenio-records-resources/blob/master/invenio_records_resources/services/records/service.py#L23). 

Before, the functions were named `{create, read, etc.}_secret_link` to avoid overwriting a record one. But now that it is a subservice... if they follow the same interface (they should) they can be renamed to `read`, `create` etc.